ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -147,25 +147,17 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         return new IngestionStatusResult(ingestionStatus);
     }
 
-    private String getFormat(IngestionProperties ingestionProperties) {
-        String format = ingestionProperties.getDataFormat();
-        if (format == null) {
-            return "csv";
-        }
-        return format;
-    }
-
-    private String getMappingReference(IngestionProperties ingestionProperties, String format) throws IngestionClientException {
+    private String getMappingReference(IngestionProperties ingestionProperties, IngestionProperties.DataFormat dataFormat) throws IngestionClientException {
         IngestionMapping ingestionMapping = ingestionProperties.getIngestionMapping();
         String mappingReference = ingestionMapping.getIngestionMappingReference();
         IngestionMapping.IngestionMappingKind ingestionMappingKind = ingestionMapping.getIngestionMappingKind();
-        if (IngestionMapping.mappingRequiredFormats.contains(format)) {
+        if (dataFormat.isMappingRequired()) {
```

Comment:
Not introduced in this PR - but are we only checking this in streaming ingestion? shouldn't this check also be in queued?

Reply:
I assume if we did a full client comparison we'd find several gaps just because someone thought of 1 at some point after others were implemented.

As for this specific point, I agree that it would only make sense to include it in each client.

Reply:
In python we check it in the ctor of ingestionproperties so it's all of the clients:
```python
        if ingestion_mapping is not None and ingestion_mapping_reference is not None:
            raise KustoMappingAndMappingReferenceError()

        if data_format.mapping_required and ingestion_mapping_reference is None and ingestion_mapping is None:
            raise KustoMissingMappingReferenceError(data_format.kusto_value)

```

At first I thought we should do it here, but it's actually a subtle bug I noticed now, because if you have streaming ingestion and you provide it with column mapping, the check will pass but streaming ingestion only supports reference.
Maybe something to add to the other PR?

Make sure that if you refactor it here you don't have the same pitfall

Reply:
I vaguely remember that StreamingIngest requires a mapping reference only, so if that's true, I'll add that restriction to `StreamingIngestClient.ingestFromStream()` now:
`Ensure.stringIsNotBlank(ingestionProperties.getIngestionMapping().getIngestionMappingReference(), "ingestionMappingReference");`

Might as well get it all working as expected before a major bump.

Reply:
It requires a mapping reference only for types that require a mapping.
You don't have to pass it for csv for example.

Reply:
Yeah I just figured that out and modified the code accordingly.